# News
- I will present  BLight, our associative structure for k-mers, at [RECOMB-seq](https://recombseq.recomb2019.org/program/), on May 4.
- I will present our ongoing work on colored De Bruijn graphs and SBT at [Helsinki Bioinformatic Day]( https://www.helsinki.fi/en/researchgroups/bioinformatics/bioinformatics-day-2019), on May 29.
- Leandro Ishi's [Comparative assessment of long-read error-correction software applied to RNA-sequencing data](https://www.biorxiv.org/content/10.1101/476622v1) is out
- We released [ELECTOR's](https://www.biorxiv.org/content/10.1101/512889v1) preprint

# Ongoing work
- [Comparative assessment of long-read error-correction software applied to RNA-sequencing data](https://www.biorxiv.org/content/10.1101/476622v1), L Ishi Soares de Lima, C Marchet, S Caboche, C Da Silva, B Istace, J-M Aury, H Touzet, R Chikhi
- [ELECTOR: Evaluator for long reads correction methods](https://www.biorxiv.org/content/10.1101/512889v1), C Marchet, P Morisse, L Lecompte, A Limasset, A Lefebvre, T Lecroq, P Peterlongo 

# Journal Publications
- [Clustering _de Novo_ by Gene of Long Reads from Transcriptomics Data](https://academic.oup.com/nar/article/47/1/e2/5107577), C Marchet, L Lecompte, C Da Silva, C Cruaud, J-M Aury, J Nicolas, P Peterlongo, Nucleic Acid Research, 2018
- [A _de novo approach_ to disentangle partner identity and function in holobiont systems](https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-018-0481-9) A Meng, C Marchet, E Corre, P Peterlongo, A Alberti, C Da Silva, P Wincker, E Pelletier, I Probert, J Decelle, S Le Crom, F Not, L Bittner, Microbiome, 2018
- [A resource-frugal probabilistic dictionary and applications in bioinformatics](https://www.sciencedirect.com/science/article/pii/S0166218X18301288), C Marchet, L Lecompte, A Limasset, L Bittner, P Peterlongo, Discrete Applied Mathematics, 2018
- [Complementarity of assembly-first and mapping-first approaches for alternative splicing annotation and differential analysis from RNAseq data](https://www.nature.com/articles/s41598-018-21770-7) C Benoit-Pilven, C Marchet, E Chautard, L Lima, M-P Lambert, G Sacomoto, A Rey, C Bourgeois, D Auboeuf, V Lacroix, Scientific Reports, 2018
- [Playing hide and seek with repeats in local and global de novo transcriptome assembly of short RNA-seq reads](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5322684/) L Lima, B Sinaimeri, G Sacomoto, H Lopez-Maestre, C Marchet, V Miele, M-F Sagot and V Lacroix Algorithms for Molecular Biology 2017

# Papers accepted to conferences
- [Indexing De Bruijn graphs with minimizers
](https://www.biorxiv.org/content/10.1101/546309v2), C Marchet, A Kerbiriou, A Limasset, accepted to RECOMB-seq 2019
- [CONSENT: Scalable self-correction of long reads with multiple sequence alignment
](https://www.biorxiv.org/content/10.1101/546630v3), P Morisse, C Marchet, A Limasset, A Lefebvre, T Lecroq, accepted to RECOMB-Seq 2019
- [A resource-frugal probabilistic dictionary and applications in (meta)genomics](https://hal.inria.fr/hal-01386744/document), C Marchet, A Limasset, L Bittner, P Peterlongo, PSC 2016
- [Navigating in a Sea of Repeats in RNA-seq without Drowning](https://link.springer.com/chapter/10.1007/978-3-662-44753-6_7), G Sacomoto, B Sinaimeri, C Marchet, V Miele, MF Sagot, V Lacroix, International Workshop on Algorithms in Bioinformatics (WABI) 2014

# Communications

## Invited talks
**2019**
- _New methodologies for the analysis of transcriptome sequences_, MAB team seminar, LIRMM Montpellier (France)

**2018**
- CARNAC-LR and C2C: de novo clustering and detection of alternative isoforms in Third Generation Sequencing transcriptomes, [Genotoul Biostats/Bioinfo day](https://bioinfo-biostat.sciencesconf.org/), INRA Toulouse (France)
- From reads to transcripts: _de novo_ methods to analyze transcriptome 2d and 3d generations sequencing data, Roscoff Biological Station seminar (France)
- A _de novo_ approach to disentangle partner identity and function in holobiont systems, Advances techniques to study and exploit the sponge and coral microbiomes Workshop, ULB Brussels (Belgium)
- A highly scalable data structure for read similarity computation and its application to marine holobionts, EEB group meeting, ULB Brussels (Belgium)
- CARNAC-LR: clustering genes expressed variants from long read RNA sequencing, team TIBS seminar, LITIS Rouen (France)

**2017**
- _De novo_ Clustering of Gene Expressed Variants in Transcriptomic Long Reads Data Sets at workshop RNA-Seq and Nanopore sequencing, Genoscope Evry (France)
- Rconnector: a resource-frugal probabilistic dictionary and applications in (meta)genomics and transcriptomics, LBEE NGS group seminar, Lyon (France)

## Workshops/conferences/seminars

**2019**
- [Survey of _k_-mer set of sets data structures for querying large collections of sequencing datasets](https://dsb2019.gitlab.io/), DSB, Dortmund (Germany)

**2018**
- [BCOOL-Trans: accurate and variant-preserving correction for RNA-seq](http://bioinfo.univ-rouen.fr/seqbio2018/programme.php), Seqbio, Rouen (France)
- [ELECTOR: EvaLuator of Error CorrectionTools for lOng Reads](http://bioinfo.univ-rouen.fr/seqbio2018/programme.php), Seqbio, Rouen (France)
- [CARNAC-LR : Clustering coefficient-based Acquisition of RNA Communities in Long Reads](https://jobim2018.sciencesconf.org/data/pages/Program_A4_JOBIM2018.pdf), JOBIM, Marseille (France)

**2017**
- _De novo_ Clustering of Gene Expression in Transcriptomic Long Reads Data Sets, Seqbio, Lille (France)
- A highly scalable data structure for read similarity computation and its application to marine holobionts, RCAM, Paris (France)
- A scaling transcriptomic approach to study holobiont data sets, 4e colloque Génomique Environnementale, Marseille (France)

**2016** 
- Rconnector: a resource-frugal probabilistic dictionary and applications in (meta)genomics and transcriptomics, SeqBio 2016, Nantes (France)

**2015** 
- kissDE : a replicate-wise and annotation-free R package for testing the association between differential variants and experimental conditions in high throughput sequencing data, SeqBio 2015, Orsay (France)
- kissDE : a package to test for differences in reads counts derived from variants in RNA sequencing data, Quatriemes rencontres R 2015, Lyon (France)

# Teaching

<table>
  <tr>
    <th>Year</th><th>Level</th><th>Topic</th>
  </tr>
  <tr>
    <td>2018/2019</td><td>L2 informatique Université de Lille</td><td>TP/TP Algorithmique et structures de données</td>
  </tr>
  <tr>
    <td>2016/2017</td><td>4ème et 5ème année école d'ingenieur (M1/M2) INSA Rennes</td><td>Modélisation et ingénierie pour le vivant</td>
  </tr>
  <tr>
    <td>2016/2017</td><td>2ème année cycle prépa INSA Rennes</td><td>TP/TP Bases de Données</td>
  </tr>
  <tr>
    <td>2016/2017</td><td>L3 ENS Rennes</td><td>Biostatistiques, programmation avec R</td>
  </tr>
    <tr>
    <td>2015/2016</td><td>L3 ENS Rennes</td><td>Biostatistiques, programmation avec R</td>
  </tr>
   <tr>
    <td>2014/2015</td><td>L1 Université Lyon 1</td><td>TD Mathématiques pour les Sciences de la Vie</td>
  </tr>
</table>


# Supervision
- 2018: Benjamin Churcheward (M2 internship), subject: Isoform prediction using long reads sequencing
- 2017: Lolita Lecompte (M2 internship), subject: Conception and evaluation of a pipeline for de novo study of long reads in transcriptomics
- 2014: Camille Sessegolo (L3 internship), subject: Improvements in the kissplice2refgenome software

# Involvement in scientific events
- Apr. 19th-24th 2018: Volunteer at RECOMB-seq and RECOMB, Paris
- Nov. 7th-8th 2016: Organisation commitee for Biological insights from raw high-throughput sequencing data - workshop Colib read
- Mar. 24th-27th 2015: Tutor at Bioinformatique pour les NGS CNRS Montpellier
- May 26th 2014: Tutor at BGE & EMBnet tutorials (Lyon): RNA-seq analysis
- Feb. 11th-13th 2014 / Nov. 26th-28th 2014: Tutor at PRABI training courses (Lyon), Analyse de donnees RNA-seq sous l environnement Galaxy
